NumLib/ODESolver/NonlinearSolver.cpp

```diff
@@ -45,6 +45,7 @@ bool NonlinearSolver<NonlinearSolverTag::Picard>::solve(
 
     bool error_norms_met = false;
 
+    MathLib::LinAlg::finalizeAssembly(x);
```

Comment:
I agree with @norihiro-w. I guess (didn't check), that the error is caused, because `finalizeAssembly()` is not called after `applyKnownSolutions()` in the time loop. Maybe the errors are fixed if you add `finalizeAssembly()` to `applyKnownSolutions()` in `TimeDiscretizedODESystem.cpp`: https://github.com/wenqing/ogs/blob/1dbf4cd153e7302cfda83d4212164b5990874981/NumLib/ODESolver/TimeDiscretizedODESystem.cpp#L34.